### PR TITLE
Fix Manual Search, Empty Title&Kodi isn't playing

### PR DESCRIPTION
**Pull request by burekas7**
* Fix the Manual Search which have not worked until now. 

* Add support when title is given as an empty string (This happens with some addons)

* Add the option for using the addon when kodi isn't playing:
** Open the Subtitles Dialog outside the player can be done by add it in the Keymap xml files.
** This fix doesn't change the normal operation as it today when kodi is playing media.
** This fix have been tested by me (burekas)
** The mainly changes are by add a condition when kodi playing or not
playing: if xbmc.Player().isPlaying()
** There is no exception when subtitle is downloaded

### CODE REVIEW

**Comment by slmosl** [comment]
great job @burekas7, fix the minor issues I've pinpointed and we'll merge

**Comment by burekas7** [comment]
Where can I see the minor issues that you have mentioned?

By the way, I'm watching now and see that there is duplicate function: I add a function "normalizeString" which was already there by the name "normalize_string".
But I see there is a difference with the encode value utf-8 vs ascii,
So what is the right choice? (I added the function with the utf-8)

Is this the only comment that you mentioned?

File: resources/lib/SubtitleHelper.py [hunk 1]
```diff
@@ -33,3 +33,23 @@ def convert_to_utf(file):
             output.write(srt_data)
     except UnicodeDecodeError:
         log(__name__, "got unicode decode error with reading subtitle data")
+
+def normalizeString(str):
```
Comment (slmosl) [inline resources/lib/SubtitleHelper.py]: use the existing `normalize_string` function

File: service.py [hunk 2]
```diff
@@ -51,11 +50,20 @@ def search(item):
...
+            else:              
+                title, year = xbmc.getCleanMovieTitle(item['title'])
+                subtitles_options = downloader.search_movie(title)
         else:
-            title, year = xbmc.getCleanMovieTitle(item['title'])
-            subtitles_options = downloader.search_movie(title)
+            item['tvshow'], item['season'], item['episode'] = check_and_parse_if_title_is_TVshow(item['title'])
+            subtitles_options = downloader.search_tvshow(item['tvshow'], int(item['season']), int(item['episode']))
+            if (subtitles_options == None): # This is a movie title
+                item['title'] = item['title'].replace("%20", "%2b") # " " to "+"
```
Comment (burekas7) [inline service.py]: From the tests I did this is required.
Without this line, if user input for example: "Avatar 2009"
item['title'] = Avatar%202009
After getCleanMovieTitle you will get: title = avatar, year = 2009, which is OK but search_movie function return None.
With this line you are getting: item['title'] = Avatar%2b2009
And after getCleanMovieTitle => title = Avatar+2009, year = (empty)
Then for unknown reason you are actually getting results from the search_movie function.

File: service.py [hunk 3]
```diff
@@ -51,11 +50,20 @@ def search(item):
...
+            if item['tvshow'] != "":
+                subtitles_options = downloader.search_tvshow(item['tvshow'], item['season'], item['episode'])
+            else:              
+                title, year = xbmc.getCleanMovieTitle(item['title'])
+                subtitles_options = downloader.search_movie(title)
         else:
-            title, year = xbmc.getCleanMovieTitle(item['title'])
-            subtitles_options = downloader.search_movie(title)
+            item['tvshow'], item['season'], item['episode'] = check_and_parse_if_title_is_TVshow(item['title'])
+            subtitles_options = downloader.search_tvshow(item['tvshow'], int(item['season']), int(item['episode']))
```
Comment (burekas7) [inline service.py]: You are right, I'll fix it.

Comment (burekas7) [inline service.py]: You are right, I'll fix it.

File: resources/lib/SubtitleHelper.py (hunk 1 above)
Comment (burekas7) [inline resources/lib/SubtitleHelper.py]: Done.

File: resources/lib/SubtitleHelper.py [hunk 4]
```diff
@@ -33,3 +33,23 @@ def convert_to_utf(file):
...
+    return unicodedata.normalize(
+        'NFKD', unicode(unicode(str, 'utf-8'))
+    ).encode('utf-8', 'ignore')
+
+def check_and_parse_if_title_is_TVshow(manualTitle):
```
Comment (burekas7) [inline resources/lib/SubtitleHelper.py]: I totally changed the implemention and now it's using regex.
It looks a little longer, but with no bugs for sure.

**Pull request merged**